hbase_sql/src/main/java/com/splicemachine/stream/accumulator/BadRecordsAccumulator.java

```diff
@@ -14,34 +14,54 @@
 
 package com.splicemachine.stream.accumulator;
 
-import org.apache.spark.AccumulableParam;
 import com.splicemachine.derby.stream.control.BadRecordsRecorder;
+import org.apache.spark.util.AccumulatorV2;
 
 /**
  *
  * Accumulator for Bad Records from bulk import.
  * <p/>
  * Uses a {@link BadRecordsRecorder} to record each bad record to a temp file. Temp files
- * will be merged {@link #addInPlace(BadRecordsRecorder, BadRecordsRecorder)}.
+ * will be merged {@link #merge(AccumulatorV2<String, BadRecordsRecorder>)}.
  *
- * @see org.apache.spark.AccumulableParam
+ * @see org.apache.spark.util.AccumulatorV2;
  *
  */
-public class BadRecordsAccumulator implements AccumulableParam<BadRecordsRecorder, String> {
+public class BadRecordsAccumulator extends AccumulatorV2<String, BadRecordsRecorder> {
+
+    private final BadRecordsRecorder badRecordsRecorder;
+
+    public BadRecordsAccumulator(BadRecordsRecorder badRecordsRecorder) {
+        this.badRecordsRecorder = badRecordsRecorder;
+    }
+
+    @Override
+    public boolean isZero() {
+        return badRecordsRecorder.getNumberOfBadRecords() == 0;
+    }
+
+    @Override
+    public AccumulatorV2<String, BadRecordsRecorder> copy() {
+        return new BadRecordsAccumulator(badRecordsRecorder);
+    }
```

Comment:
if i understand correctly, you might as well return **this** here:
BadRecordsAccumulator only has one internal state, namely the badRecordsRecorder. With this code, every "copy" points to the same badRecordsRecorder, so will have a shared state, making copies not really copies. e.g. everything you would add or merge to one BadRecordsAccumulator, you would also see in the copy. Not sure if this is intended.

Reply:
you are right, thanks. I've improved the copy method, added reset for BadRecordsRecorder as well